config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ServiceEntityBuilder.java

```diff
@@ -67,24 +71,38 @@ private Entity buildServiceEntity(Bundle bundle, String servicePath, Service ser
         serviceDetailElement.appendChild(buildServiceMappings(service, document));
 
         if (service.getProperties() != null) {
-            buildAndAppendPropertiesElement(service.getProperties()
-                            .entrySet()
-                            .stream()
-                            .collect(Collectors
-                                    .toMap(p -> "property." + p.getKey(), p -> p.getKey().startsWith(PREFIX_ENV) ? "SERVICE_PROPERTY_" + p.getKey() : p.getValue())),
+            Map<String, Object> properties = service.getProperties()
+                    .entrySet()
+                    .stream()
+                    .collect(Collectors
+                            .toMap(p -> "property." + p.getKey(), p -> p.getKey().startsWith(PREFIX_ENV) ? "SERVICE_PROPERTY_" + p.getKey() : p.getValue()));
+
+            if(isSoapService) {
+                properties.put(KEY_VALUE_SOAP, true);
+                properties.put(KEY_VALUE_SOAP_VERSION, wsdlBean.getSoapVersion());
+            }
+            buildAndAppendPropertiesElement(properties,
                     document, serviceDetailElement);
         }
 
         Element serviceElement = createElementWithAttribute(document, SERVICE, ATTRIBUTE_ID, id);
         serviceElement.appendChild(serviceDetailElement);
 
         Element resourcesElement = document.createElement(RESOURCES);
-        Element resourceSetElement = createElementWithAttribute(document, RESOURCE_SET, "tag", "policy");
-        Element resourceElement = createElementWithAttribute(document, RESOURCE, "type", "policy");
-        resourceElement.setTextContent(documentFileUtils.elementToString(policy.getPolicyDocument()));
+        Element policyResourceSetElement = createElementWithAttribute(document, RESOURCE_SET, ATTRIBUTE_TAG, TAG_VALUE_POLICY);
+        Element policyResourceElement = createElementWithAttribute(document, RESOURCE, ATTRIBUTE_TYPE, TAG_VALUE_POLICY);
+        policyResourceElement.setTextContent(documentFileUtils.elementToString(policy.getPolicyDocument()));
+        policyResourceSetElement.appendChild(policyResourceElement);
+        resourcesElement.appendChild(policyResourceSetElement);
+
+        if(isSoapService) {
+            Element wsdlResourceSetElement = createElementWithAttributes(document, RESOURCE_SET, ImmutableMap.of(ATTRIBUTE_TAG, TAG_VALUE_WSDL, ATTRIBUTE_ROOT_URL, wsdlBean.getRootUrl()));
```

Comment:
wsdlBean.getRootUrl() will return null if the jsonIgnore is set for the rootUrl property in a situation that is export and then build

Reply:
that's was I was having problem at.. :)